packages/@aws-cdk/aws-config/README.md

```diff
@@ -26,39 +30,150 @@ following resources per region:
+
+Checks whether your CloudFormation stack's actual configuration differs, or has drifted,
+from it's expected configuration. 
+
+```ts
+import * as config from '@aws-cdk/aws-config';
+import * as cdk from '@aws-cdk/aws-cdk';
+
+// compliant if stack's status is 'IN_SYNC'
+// non-compliant if the stack's drift status is 'DRIFTED'
+new config.CloudFormationStackDriftDetectionCheck(stack, 'Drift', {
+  ownStackOnly: true, // checks only the stack containing the rule
+});
+```
+
+##### CloudFormation Stack notifications
+
+Checks whether your CloudFormation stacks are sending event notifications to a SNS topic.
+
+```ts
+import * as config from '@aws-cdk/aws-config';
+import * as cdk from '@aws-cdk/aws-cdk';
+
+// topics to which CloudFormation stacks may send event notifications
+const topic1 = new sns.Topic(stack, 'AllowedTopic1');
+const topic2 = new sns.Topic(stack, 'AllowedTopic2');
+
+// non-compliant if CloudFormation stack does not send notifications to 'topic1' or 'topic2'
+new config.CloudFormationStackNotificationCheck(this, 'NotificationCheck', {
+  topics: [topic1, topic2],
+})
+```
 
 #### Custom rules
 
-To set up a custom rule, define a `CustomRule` and specify the Lambda Function to run and the trigger types:
+You can develop custom rules and add them to AWS Config. You associate each custom rule with an
+AWS Lambda function, which contains the logic that evaluates whether your AWS resources comply
+with the rule.
+
+#### Triggers
+
+AWS Lambda executes functions in response to events that are published by AWS Services.
+The function for a custom Config rule receives an event that is published by AWS Config,
+and the function then uses data that it receives from the event and that it retrieves from
+the AWS Config API to evaluate the compliance of the rule.
+
+##### Configuration Changes
+
+AWS Config will invoke your Lambda function when it detects a configuration change for a resource.
+Example function for [evaluations triggered by configuration changes](https://docs.aws.amazon.com/config/latest/developerguide/evaluate-config_develop-rules_nodejs-sample.html#event-based-example-rule)
 
 ```ts
-new CustomRule(this, 'CustomRule', {
-  lambdaFunction: myFn,
+import * as config from '@aws-cdk/aws-config';
+
+new config.CustomRule(this, 'CustomRule', {
+  lambdaFunction: evalComplianceFn,
   configurationChanges: true,
-  periodic: true
 });
 ```
 
-#### Restricting the scope
+##### Periodic
+
+AWS Config will invoke a function periodically at the frequency you specify.
+Example function for [periodic evaluations](https://docs.aws.amazon.com/config/latest/developerguide/evaluate-config_develop-rules_nodejs-sample.html#periodic-example-rule)
+
+```ts
+import * as config from '@aws-cdk/aws-config';
+
+new config.CustomRule(this, 'CustomRule', {
+  lambdaFunction: evalComplianceFn,
+  periodic: true,
+  maximumExecutionFrequency: config.MaximumExecutionFrequency.SIX_HOURS, // default is 24 hours
+});
+```
```

Comment:
I think I'd delete both this sections, and just retain the links to the examples. So:

```
AWS Lambda executes functions in response to events...

Evaluations can be triggered by configuration changes, periodically, ....

(Example with both props set to true)

When the trigger for a rule occurs...

The AWS documentation has examples of Lambda functions [triggered by configuration changes] and [triggered perodically].
```

I think that gets all of the same ideas, but in much less space. WDYT?

Reply:
i like it. took a run at it.